### PR TITLE
Dynamic Adjustments Part 2

### PR DESCRIPTION
**Pull request by nednaZ**
# About The Pull Request

Ports:
https://github.com/tgstation/tgstation/pull/59463
https://github.com/tgstation/tgstation/pull/60988
https://github.com/tgstation/tgstation/pull/63305
https://github.com/tgstation/tgstation/pull/60281 (Except for disease)
https://github.com/tgstation/tgstation/pull/62258

Changes the intercept message to be more flavorful.

Clamps the threat level to between 75% player pop and 200% player pop.

Logs increases to Dynamic threat budget.

Slightly reduces the weight of latejoin traitors. (From 7 to 4)

Increases the Weight (2 to 4), decreases the Cost (40 to 20) and decreases Minimum Players (35 to 30) of Revolution Latejoins.

Restores Heretics to Latejoin and Roundstart.

Heretics now have a lower number of sacrifice objectives. (From 2-6 to 1-3)

Heretics now have a chance to get a Steal objective.

Dynamic Abductors may no longer get spawned  in solo by mistake.

Midround Traitor chance reduced. (From 7 to 5)

Midround Wizard weight increased (From 1 to 3)
Midround Wizards are now a HIGH_IMPACT_RULESET and will not repeat.

Midround Nuclear Operative Weight reduced. (From 5 to 4)

Blob Weight (4 to 3) and Cost (10 to 25) changed.

Xenomorph Cost increased (10 to 25)

Nightmare Cost increased (10 to 15)

Abductor Cost increased (10 to 15)

Space Pirates added to Dynamic.

Space Pirates have been given an update, with ship names and support for multiple types of pirates existing.

Revenant added to Dynamic.

Obsessed added to Dynamic.

Space Dragon early start changed to 40 minutes (From 50 minutes)

Roundstart Traitor cost increased (7 to 8)

Blood Brother Weight (4 to 2), Cost(15 to 13) and Scaling Cost(15 to 13) adjusted.

Changeling Weight(3 to 4) and Cost(15 to 17) adjusted.

Roundstart Wizard Weight(2 to 5) and Cost(40 to 30) adjusted.

Blood and Clock Cult Weight and Cost(30 to 25) adjusted.

Nuclear Operatives Weight(3 to 4) and Cost(50 to 25) adjusted.

## Why It's Good For The Game

More variety is good.
This should give us more options rather than plain traitor/BB all the time.

## Changelog

:cl:
add: Changes the intercept messages to be more flavorful.
balance: Clamps the threat level to between 75% player pop and 200% player pop.
admin: Logs increases to Dynamic threat budget.
balance: Slightly reduces the weight of latejoin traitors. (From 7 to 4)
balance: Increases the Weight (2 to 4), decreases the Cost (40 to 20) and decreases Minimum Players (35 to 30) of Revolution Latejoins.
add: Restores Heretics to Latejoin and Roundstart.
add: Space Pirates added to Dynamic.
add: Space Pirates have been given an update, with ship names and support for multiple types of pirates existing.
add: Revenant added to Dynamic.
add: Obsessed added to Dynamic.
balance: Heretics now have a lower number of sacrifice objectives. (From 2-6 to 1-3)
add: Heretics now have a chance to get a Steal objective.
fix: Dynamic Abductors may no longer get spawned in solo by mistake.
fix: Lone Operatives should now spawn. SECURE THAT DAMN DISK
balance: Midround Traitor chance reduced. (From 7 to 5)
balance: Midround Wizard weight increased (From 1 to 3)
balance: Midround Wizards are now a HIGH_IMPACT_RULESET and will not repeat.
balance: Midround Nuclear Operative Weight reduced. (From 5 to 4)
balance: Blob Weight (4 to 3) and Cost (10 to 25) changed.
balance: Xenomorph Cost increased (10 to 25)
balance: Nightmare Cost increased (10 to 15)
balance: Abductor Cost increased (10 to 15)
balance: Space Dragon early start changed to 40 minutes (From 50 minutes)
balance: Roundstart Traitor cost increased (7 to 8)
balance: Blood Brother Weight (4 to 2), Cost(15 to 13) and Scaling Cost(15 to 13) adjusted.
balance: Changeling Weight(3 to 4) and Cost(15 to 17) adjusted.
balance: Roundstart Wizard Weight(2 to 5) and Cost(40 to 30) adjusted.
balance: Blood and Clock Cult Weight and Cost(30 to 25) adjusted.
balance: Nuclear Operatives Weight(3 to 4) and Cost(50 to 25) adjusted.
/:cl:

### CODE REVIEW

**Comment by wraith-54321** [comment]
this still has old obsessed cost of 10, I would honestly say TGs 3 cost obsessed would be fine

**Comment by nednaZ** [comment]
I'm willing to meet in the middle with a cost of 5, since 3 seems far too small of a threat cost

File: code/game/gamemodes/dynamic/dynamic_rulesets_midround.dm [hunk 1]
```diff
@@ -501,3 +507,122 @@
...
+/datum/dynamic_ruleset/midround/obsessed
+	name = "Obsessed"
+	antag_datum = /datum/antagonist/obsessed
+	antag_flag = ROLE_OBSESSED
+	restricted_roles = list("Cyborg", "AI", "Positronic Brain")
+	enemy_roles = list("Security Officer", "Detective", "Head of Security", "Captain")
+	required_enemies = list(2,2,1,1,1,1,1,0,0,0)
+	required_candidates = 1
+	weight = 3
+	cost = 10
```
Comment (nednaZ) [inline code/game/gamemodes/dynamic/dynamic_rulesets_midround.dm:603]:
```suggestion
	cost = 5
```

**Comment by nednaZ** [comment]
I'll make changes as I get more player feedback, but I haven't seen any complaints about the system as it stands after a good amount of testmerging.

**Pull request merged**